### ISSUE TITLE
Instantiate command should be on the smart contract rather than the channel

### ISSUE THREAD

**Issue by lehors** [IBM-Blockchain/blockchain-vscode-extension#549]
The current UI requires me to click on the channel name, select "Instantiate contract" and then select the contract. Instead, I should be able to directly click on the contract under the channel and select "Instantiate".

**Comment by caroline-church** [comment]
@lehors a smart contract only appears under a channel once it has been instantiated. You can upgrade by right clicking on a smart contract

**Comment by lehors** [comment]
That's not what I'm seeing. Here is what my screen looks like after install, and *before* instantiate.
<img width="991" alt="screen shot 2019-02-08 at 3 42 44 pm" src="https://user-images.githubusercontent.com/6464618/52485305-d7711500-2bb8-11e9-95fe-20ed6a55a99d.png">

**Comment by caroline-church** [comment]
Ahhh that is when it is installed on a peer, sorry the description sounded like you meant directly under the channel. This is a question for design so tagging @EdMoffatt

**Comment by EdMoffatt** [comment]
Worth taking a look at this in the context of the latest extension version, as the local_fabric ops panel has shuffled around a little.  Here's what we see now:

![image](https://user-images.githubusercontent.com/7454304/53236889-49ebf580-368d-11e9-8880-2b3ca27d5581.png)

If we allowed right-click of the contracts in the installed list, we'd need to do some logic to decide what can be in the right-click menu, unless we want to offer users actions that definitely won't work (which is worth avoiding if possible!).  So in this example...

- YUNOFEEDBACK cannot be instantiated
- fabcar cannot be instantiated
- any of the feedback contracts could be instantiated

Then consider if there was a second channel (not available in the extension yet, but something we'd like to add in future).  Suppose that channel had feedback@v1.0.0 instantiated.  That complicates things further, as the instantiate action is valid or invalid on any of the smart contracts on the list, depending which channel you pick!

- - -

Whilst the extension today only allows a single channel, a lot of the UX decisions have been made considering the future when there could be multiple, with a single peer able to have joined multiple channels.

With a single channel, and peers shown as children of the channel, I agree your suggestion would be logical for a user to try and a good thing for us to implement!
However, for the same reason we've removed that heirarchy of peers-inside-channels from the view in preparation for more complex local_fabric configurations, I don't think a right-click-installed-to-instantiate would be quite as clean as suggested.

- - -

End thought: we should come back to these operations and review once we have the ability to reconfigure the local_fabric.  Right now, I don't think this is any better than the existing behaviour (and I don't like having too many multiple ways of doing things, as it means each user could develop a different mental model of how things work)
